lib/utils/logger.ts

```diff
@@ -19,6 +19,13 @@ import _ = require('lodash');
 import { EOL as eol } from 'os';
 import { StreamLogger } from 'resin-stream-logger';
 
+/**
+ * General purpose logger class with support for log streams and colours.
+ * Call `Logger.getLogger()` to retrieve a global shared instance of this
+ * class. The `new Logger()` pattern is not recommended because it may lead
+ * to Node printing "MaxListenersExceededWarning" warning messages to the
+ * console.
+ */
 class Logger {
```

Comment:
> _I wonder how could we preserve some of those "why"-style comments _in_ the code_

Thanks for the review! I've added some JSDoc comments to the Logger class and to the `getLogger()` method. Also, I've added the `protected` modified to the Logger `constructor`, which forces the use of `getLogger()` (preventing the `new Logger()` usage, at least in Typescript code). I hope this is not a step too far...